src/vampyr/trees/trees.h

```diff
@@ -124,140 +177,47 @@ template <int D> void trees(pybind11::module &m) {
-                add(-1.0, *out, vec);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__imul__",
-            [](FunctionTree<D> *out, double c) {
-                out->rescale(c);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__mul__",
-            [](FunctionTree<D> *inp_a, FunctionTree<D> *inp_b) {
-                auto out = std::make_unique<FunctionTree<D>>(inp_a->getMRA());
-                FunctionTreeVector<D> vec;
-                vec.push_back({1.0, inp_a});
-                vec.push_back({1.0, inp_b});
-                build_grid(*out, vec);
-                build_grid(*out, 1);
-                multiply(-1.0, *out, vec);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__imul__",
-            [](FunctionTree<D> *out, FunctionTree<D> *inp) {
-                refine_grid(*out, *inp);
-                refine_grid(*out, 1);
-                out->multiply(1.0, *inp);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__truediv__",
-            [](FunctionTree<D> *inp, double c) {
-                auto out = std::make_unique<FunctionTree<D>>(inp->getMRA());
-                FunctionTreeVector<D> vec;
-                vec.push_back({1.0 / c, inp});
-                build_grid(*out, vec);
-                add(-1.0, *out, vec);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__itruediv__",
-            [](FunctionTree<D> *out, double c) {
-                out->rescale(1.0 / c);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__pow__",
-            [](FunctionTree<D> *inp, double c) {
-                auto out = std::make_unique<FunctionTree<D>>(inp->getMRA());
-                copy_grid(*out, *inp);
-                copy_func(*out, *inp);
-                refine_grid(*out, 1);
-                out->power(c);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__ipow__",
-            [](FunctionTree<D> *out, double c) {
-                refine_grid(*out, 1);
-                out->power(c);
-                return out;
-            },
-            py::is_operator());
+        .def("__pos__", &impl__pos__<D>, py::is_operator())
+        .def("__neg__", &impl__neg__<D>, py::is_operator())
+        .def("__add__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__add__<D>), py::is_operator())
+        .def("__iadd__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__add__<D>), py::is_operator())
+        .def("__sub__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__sub__<D>), py::is_operator())
+        .def("__isub__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__sub__<D>), py::is_operator())
```

Comment:
The `overload_cast` is not needed here, since you have only one version of each magic method.

Reply:
Or: I think it's not needed!